Makefile

```diff
@@ -464,7 +464,7 @@ world: coldstart
 
 # Compile also native code compiler and libraries, fast
 .PHONY: world.opt
-world.opt: coldstart
+world.opt: checknative coldstart
```

Comment:
If `checknative` produces an error, under `-j` `coldstart` is already permitted to have started, and it won't immediately stop and `make world.opt` will do work it was not supposed to do.

My suggestion is to add `$(MAKE) coldstart` as the first line of the recipe and simply remove the `coldstart` from the dependencies, which I think is exactly the same effect as `coldstart` is `.PHONY`

Reply:
Thanks for the explanation: now I got it!  Will change as suggested.